src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Kafka/KafkaProduceSyncIntegration.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Kafka
+{
+    /// <summary>
+    /// Confluent.Kafka Producer.Produce calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "Confluent.Kafka",
+        TypeName = "Confluent.Kafka.Producer`2",
+        MethodName = "Produce",
+        ReturnTypeName = ClrNames.Void,
+        ParameterTypeNames = new[] { KafkaConstants.TopicPartitionTypeName, KafkaConstants.MessageTypeName, KafkaConstants.ActionOfDeliveryReportTypeName },
+        MinimumVersion = "1.4.0",
+        MaximumVersion = "1.*.*",
+        IntegrationName = KafkaConstants.IntegrationName)]
+    public class KafkaProduceSyncIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TTopicPartition">Type of the TopicPartition</typeparam>
+        /// <typeparam name="TMessage">Type of the message</typeparam>
+        /// <typeparam name="TDeliveryHandler">Type of the delivery handler action</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="topicPartition">TopicPartition instance</param>
+        /// <param name="message">Message instance</param>
+        /// <param name="deliveryHandler">Delivery Handler instance</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TTopicPartition, TMessage, TDeliveryHandler>(TTarget instance, TTopicPartition topicPartition, TMessage message, TDeliveryHandler deliveryHandler)
+            where TMessage : IMessage
+        {
+            // manually doing duck cast here so we have access to the _original_ TopicPartition type
+            // as a generic parameter, for injecting headers
+            Scope scope = KafkaHelper.CreateProducerScope(
+                Tracer.Instance,
+                topicPartition.DuckCast<ITopicPartition>(),
+                isTombstone: message.Value is null,
+                finishOnClose: deliveryHandler is null);
```

Comment:
If the span doesn't get closed when the Scope is disposed, who ensures it gets closed?

Reply:
Resolving, I hadn't finished reading to the ProduceSync DeliveryHandler integration